src/nearcache/RepairingHandler.ts

```diff
@@ -36,10 +35,9 @@ export class RepairingHandler {
         this.nearCache = nearCache;
         this.name = name;
         this.partitionService = partitionService;
-        this.partitionCount = this.partitionService.getPartitionCount();
         this.localUuid = localUuid;
         this.containers = [];
-        for (let i = 0; i < this.partitionCount; i++) {
+        for (let i = 0; i < this.partitionService.getPartitionCount(); i++) {
```

Comment:
Say hello to my little bug! :)

Reply:
That's strange. In this particular class (`RepairingHandler`) your change shouldn't make any difference as `partitionCount` field is initialized with `partitionService.getPartitionCount()` and then used in the loop condition in the same synchronous code. Maybe it's changes in `RepairingTask` that make the difference? In any case, did you find out the reason why this test failure wasn't showing up before this PR? Maybe, the reason is in the changed service initialization order?

Reply:
Ok, right. I should not change RepairingHandler. Bug is in the ReparingTask.

You can see in the commit, 

I was setting partition count in the constructor
```python
this.partitionCount = this.partitionService.getPartitionCount();
```

but, then its value is contant and whenever isAboveMaxToleratedMissCount is called its value is the value set during construction which is wrong.

Reply:
Now I see that in `master` the usage in for loop is `this.partitionCount`, rather than using  ` this.partitionService.getPartitionCount()` in the for loop. I will try to understand root cause of this, by looking it again. But, this change fixed the issue for now. It might be due to service initialization order

Reply:
Thanks for the explanation. Looks like the reason is in https://github.com/hazelcast/hazelcast-nodejs-client/pull/797/files#diff-2ce95a5b51b2a2fd60ce960ef8c97547152dc185d94deebab00db0660233354aR192

Before this PR `ProxyManager` was calling `client.getRepairingTask()` in lazy fashion and now the call happens during the client initialization.

Reply:
So the partition count is wrong when it is constructed.  It needs to be called again in for loops? Which means RepairingTask should stay as it is.

Reply:
> So the partition count is wrong when it is constructed.

Yes, the partition count is fetched from the cluster once the client on successful auth.

Reply:
Your fix for `ReparingTask` does the job (I'd only add some comments to avoid confusion for future readers). As an alternative approach, we could keep lazy initialization logic for `client.getRepairingTask()` calls, but I don't have a strong opinion here.

Reply:
It is called in `removeNearCacheInvalidationListener` and `createNearCacheEventHandler`. Passing a closure as before is an option but what do you think? Or just passing the client to ProxyManager is another one. (With an interface just like connection manager)

Reply:
Passing a closure (i.e. factory function with some side effects) sound like a better option to me as from the `ProxyManager`'s perspective it's a getter, rather than a hook/callback.

Reply:
BTW, `RepairingHandler.ts` is better this way, as we do not need the `partitionCount` variable anywhere else than the constructor. Can stay like this.